runtime/doc/options.txt

```diff
@@ -5227,11 +5227,12 @@ A jump table for the options with a short description can be found at |Q_op|.
 	Note that such processing is done after |:set| did its own round of 
 	unescaping, so to keep yourself sane use |:let-&| like shown above.
 							*shell-powershell*
-	To use powershell: >
+	To use PowerShell: >
 		let &shell = has('win32') ? 'powershell' : 'pwsh'
-		set shellquote= shellpipe=\| shellxquote=
-		set shellcmdflag=-NoLogo\ -NoProfile\ -ExecutionPolicy\ RemoteSigned\ -Command
-		set shellredir=\|\ Out-File\ -Encoding\ UTF8
+		set shellquote= shellxquote=
+		set shellcmdflag=-NoLogo\ -NoProfile\ -ExecutionPolicy\ RemoteSigned\ -Command\ [Console]::InputEncoding=[Console]::OutputEncoding=[System.Text.Encoding]::UTF8;
```

Comment:
Need a test to justify putting this in the docs or reference because the option value is so long.. Existing tests didn't need to change default input/output encoding of powershell.

Reply:
I already added this to set_shell_powershell() so it should be tested everywhere that helper is used, no?

I came up with this config by going through all of PS2, 5, and 7 and this is the most consistent one that worked for all of those. Something that looked like a one line doc fix turned out to be a really difficult problem heh.

Reply:
I think I'll need help here. I tried writing something involving unicode chars before while working on this but I could not find a correct way to enter either input or reference values or both. There's also a few test harness styles as far as I can see (screen content tests, then the ones that seem to be getting buffer contents directly, etc), which one I should be using? Is there one that's close to what I would be writing here?

Reply:
Screen test is fine for validating powershell output in the buffer.

I'd start with `system()`, `systemlist`, `jobstart()`) which can accept stdin for unicode characters and can pass its output to Vimscript  variables so that you don't need to use screen tests.
The tests should set the shell options directly to assert the expected results on different input/output encodings.

Reply:
@janlazo PTAL; see new test in eval/system_spec.lua. I looked at other tests and some of them do seem to verify default vim config (such as utf8 encoding), so first part my test uses and verifies the helper config, then ensures that config actually does something.

Reply:
That's fine. Create a similar test for `systemlist()` and I'll merge.

Reply:
Done, PTAL

Reply:
Re: odd newline after the character, apparently that's what systemlist does on Windows? https://vim-jp.org/vimdoc-en/eval.html#systemlist()

Reply:
Yes, crlf is not converted to lf for systemlist.